lib/service/navigation_service.dart

```diff
@@ -199,18 +200,26 @@ class NavigationService {
         navigatorKey.currentContext != null) {
       final metricClient = injector.get<MetricClientService>();
       metricClient.timerEvent(MixpanelEvent.cancelContact);
-      await UIHelper.showInfoDialog(
-        navigatorKey.currentContext!,
-        'contacting'.tr(),
-        'contact_with_dapp'.tr(),
-        closeButton: "cancel_dialog".tr(),
-        isDismissible: true,
-        autoDismissAfter: 20,
-        onClose: () {
-          metricClient.addEvent(MixpanelEvent.cancelContact);
-          hideInfoDialog();
-        },
-      );
+      willShowContacting = true;
+      await UIHelper.showLoadingIndicator(navigatorKey.currentContext!);
+      Future.delayed(const Duration(seconds: 4), () async {
+        hideInfoDialog();
+        if (willShowContacting) {
+          await UIHelper.showInfoDialog(
```

Comment:
since we add the delay, could you please check `mounted` and `context != null` again. I suggest to split `IHelper.showInfoDialog()` into a new child function in navigation service.